pkg/agent/controller/networkpolicy/multicast.go

```diff
@@ -0,0 +1,332 @@
+package networkpolicy
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	apitypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/interfacestore"
+	"antrea.io/antrea/pkg/agent/openflow"
+	"antrea.io/antrea/pkg/agent/types"
+	crdv1beta "antrea.io/antrea/pkg/apis/controlplane/v1beta2"
+	"antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	"antrea.io/antrea/pkg/util/ip"
+)
+
+type ruleType int
+
+type ruleEvent uint8
+
+const (
+	unicast   ruleType = 0
+	igmp      ruleType = 1
+	multicast ruleType = 2
+
+	ruleAdd    ruleEvent = 1
+	ruleUpdate ruleEvent = 2
+	ruleDelete ruleEvent = 3
+
+	groupAddressIndex = "groupAddressIndex"
+)
+
+type mcastItem struct {
+	groupAddress net.IPNet
+	ruleIDs      map[string]*uint16
+}
+
+type mcastIGMPRule struct {
+	ruleID       string
+	priority     *uint16
+	groupAddress []string
+	event        ruleEvent
+}
+
+type multicastController struct {
```

Comment:
If it is only for IGMP packets, please add comments before multicastController to make it clear.

Reply:
done